pymongo/logger.py

```diff
@@ -0,0 +1,91 @@
+from __future__ import annotations
+
+import enum
+import os
+from typing import Any
+
+from bson import UuidRepresentation, json_util
+from bson.json_util import JSONOptions
+
+
+class LogMessageStatus(str, enum.Enum):
+    STARTED = "Command started"
+    SUCCEEDED = "Command succeeded"
+    FAILED = "Command failed"
+
+
+DEFAULT_DOCUMENT_LENGTH = 1000
+_SENSITIVE_COMMANDS = [
+    "authenticate",
+    "saslStart",
+    "saslContinue",
+    "getnonce",
+    "createUser",
+    "updateUser",
+    "copydbgetnonce",
+    "copydbsaslstart",
+    "copydb",
+]
+_HELLO_COMMANDS = ["hello", "ismaster", "isMaster"]
+_REDACTED_FAILURE_FIELDS = ["code", "codeName", "errorLabels"]
+_DOCUMENTS = ["command", "reply", "failure"]
+_JSON_OPTIONS = JSONOptions(uuid_representation=UuidRepresentation.STANDARD)
+
+
+class LogMessage:
+    def __init__(self, **kwargs: Any):
+        self.kwargs = kwargs
+
+        self._process_documents()
```

Comment:
This should be delayed to __str__ for performance reasons. No need to go through all the effort of redacting info that will never be serialized.